Libplanet.RocksDBStore/RocksDBStore.cs

```diff
@@ -0,0 +1,827 @@
+        public override void ForkBlockIndexes(
+            Guid sourceChainId,
+            Guid destinationChainId,
+            HashDigest<SHA256> branchPoint)
+        {
+            LiteCollection<HashDoc> srcColl = IndexCollection(sourceChainId);
+            LiteCollection<HashDoc> destColl = IndexCollection(destinationChainId);
+
+            var genesisHash = IterateIndexes(sourceChainId, 0, 1).First();
+            destColl.InsertBulk(srcColl.FindAll()
+                .TakeWhile(i => !i.Hash.Equals(branchPoint)).Skip(1));
+            if (!branchPoint.Equals(genesisHash))
+            {
+                AppendIndex(destinationChainId, branchPoint);
+            }
+        }
+
+        /// <inheritdoc/>
+        public override IEnumerable<string> ListStateKeys(Guid chainId)
+        {
+            string collId = StateRefId(chainId);
+            return _liteDb.GetCollection<StateRefDoc>(collId)
+                .Find(Query.All(nameof(StateRefDoc.StateKey), Query.Ascending))
+                .Select(doc => doc.StateKey)
+                .ToImmutableHashSet();
+        }
+
+        /// <inheritdoc/>
+        public override IImmutableDictionary<string, IImmutableList<HashDigest<SHA256>>>
+            ListAllStateReferences(
+                Guid chainId,
+                long lowestIndex = 0,
+                long highestIndex = long.MaxValue)
+        {
+            string collId = StateRefId(chainId);
+            LiteCollection<StateRefDoc> coll = _liteDb.GetCollection<StateRefDoc>(collId);
+
+            Query query = Query.And(
+                Query.All(nameof(StateRefDoc.BlockIndex)),
+                Query.Between(nameof(StateRefDoc.BlockIndex), lowestIndex, highestIndex)
+            );
+
+            IEnumerable<StateRefDoc> stateRefs = coll.Find(query);
+            return stateRefs
+                .GroupBy(stateRef => stateRef.StateKey)
+                .ToImmutableDictionary(
+                    g => g.Key,
+                    g => (IImmutableList<HashDigest<SHA256>>)g
+                        .Select(r => r.BlockHash).ToImmutableList()
+                );
+        }
+
+        /// <inheritdoc/>
+        public override void StageTransactionIds(IImmutableSet<TxId> txids)
+        {
+            StagedTxIds.InsertBulk(
+                txids.Select(txid => new StagedTxIdDoc { TxId = txid, })
+            );
+        }
+
+        /// <inheritdoc/>
+        public override void UnstageTransactionIds(ISet<TxId> txids)
+        {
+            StagedTxIds.Delete(tx => txids.Contains(tx.TxId));
+        }
+
+        /// <inheritdoc/>
+        public override IEnumerable<TxId> IterateStagedTransactionIds()
+        {
+            IEnumerable<StagedTxIdDoc> docs = StagedTxIds.FindAll();
+            return docs.Select(d => d.TxId).Distinct();
+        }
+
+        /// <inheritdoc/>
+        public override IEnumerable<TxId> IterateTransactionIds()
+        {
+            var it = _rocksDb.NewIterator();
```

Comment:
it would be better if we specify the type of `it`. (I can't suggest because I have no idea about the return value of `NewIterator()` 😅 )

Reply:
Specified variables types in 9c83e1a.